### PR TITLE
 DAOS-1704 test: Port tests to apricot subclasses

### PR DESCRIPTION
**Pull request by brianjmurrell**

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Apologies for the new PR on this [already reviewed PR](https://github.com/daos-stack/daos/pull/377) but due to a merging mishap, no more commits could be pushed to that previous PR.  The only [new changes]( https://github.com/daos-stack/daos/pull/389/commits/36df2ddd0de2b523112874e0d02cccb8576c7782) here is @skirvan44's suggestion to use `host{list,file}_servers` instead of the bare `host{list,file}`.

File: src/tests/ftest/container/basic_tx_test.py [hunk 1]
```diff
@@ -25,75 +25,20 @@
...
-from daos_api import DaosContext, DaosPool, DaosContainer, DaosApiError
+from daos_api import DaosPool, DaosContainer, DaosApiError
 
-class BasicTxTest(Test):
+class BasicTxTest(TestWithServers):
     """
     A very simple test verifying the use of transactions.  This test was
     butchered in the transition from epoch to transaction.  Need to revisit
     and make more useful when transactions are fully working.
+    :avocado: recursive
```
Comment (skirvan44) [inline src/tests/ftest/container/basic_tx_test.py]: what is this?

File: src/tests/ftest/container/container_async.py [hunk 2]
```diff
@@ -57,52 +47,26 @@ def cb_func(event):
...
-                                                        self.workdir)
-        print("Host file is: {}".format(self.hostfile))
-
-        self.agent_sessions = AgentUtils.run_agent(self.basepath, self.hostlist)
-        server_utils.run_server(self.hostfile, self.server_group, self.basepath)
-        time.sleep(10)
-
     def tearDown(self):
         try:
             if self.pool is not None and self.pool.attached:
```
Comment (skirvan44) [inline src/tests/ftest/container/container_async.py]: Why is this necessary?

File: src/tests/ftest/container/delete.py [hunk 3]
```diff
@@ -22,37 +22,22 @@
...
-        self.hostlist = None
-        with open('../../../.build_vars.json') as build_file:
-            build_paths = json.load(build_file)
-        self.basepath = os.path.normpath(build_paths['PREFIX'] + "/../")
-        self.server_group = self.params.get("server_group", '/server/',
-                                            'daos_server')
+        super(DeleteContainerTest, self).setUp()
 
         # parameters used in pool create
         self.createmode = self.params.get("mode",
```
Comment (skirvan44) [inline src/tests/ftest/container/delete.py]: confused, this is staying?  Why?

File: src/tests/ftest/daos_test/daos_core_test.py [hunk 4]
```diff
@@ -23,57 +23,24 @@
...
                                             '/run/daos_tests/Tests/*')
 
-        # get paths from the build_vars generated by build
-        with open('../../../.build_vars.json') as build_vars:
-            build_paths = json.load(build_vars)
-        self.basepath = os.path.normpath(build_paths['PREFIX'] + "/../")
-
-        self.server_group = self.params.get("server_group", '/server/',
-                                            'daos_server')
-        self.daos_test = self.basepath + '/install/bin/daos_test'
```
Comment (skirvan44) [inline src/tests/ftest/daos_test/daos_core_test.py]: aren't some of these unique to this test?

File: src/tests/ftest/container/basic_tx_test.py (hunk 1 above)
Comment (brianjmurrell) [reply src/tests/ftest/container/basic_tx_test.py]: It's a requirement of [subclassing Avocado](https://avocado-framework.readthedocs.io/en/61.0/SubclassingAvocado.html) and is apparently the defaul in more recent versions, so once we get a newer LTS, this can go away.

File: src/tests/ftest/container/container_async.py (hunk 2 above)
Comment (brianjmurrell) [reply src/tests/ftest/container/container_async.py]: I'm not sure what you are referring to.  The line of code you are commenting on is untouched by the patch.

File: src/tests/ftest/container/delete.py (hunk 3 above)
Comment (brianjmurrell) [reply src/tests/ftest/container/delete.py]: Because the base classes don't (currently) do any fetching of the `mode` (or `size`, or other attributes used sporadically across the tests) from the params as it's not clear that this is universal across all tests for a given base class.

We may decide on a set of params that should be used for all tests and implement them consistently in the base class and tests in the future.  But I didn't want this patch to be changing the substance of the tests as they exist currently.

File: src/tests/ftest/daos_test/daos_core_test.py (hunk 4 above)
Comment (brianjmurrell) [reply src/tests/ftest/daos_test/daos_core_test.py]: Not entirely sure what you mean, but I don't think so.  This test does run and pass with this patch.
